### DISCUSSION TITLE
Speedybee F405 V3 can't save Dshot300 or Dshot600

### DISCUSSION THREAD

**Discussion by tang520-tang** [iNavFlight/inav#8910] [General]
![Snipaste_2023-03-24_14-50-27](https://user-images.githubusercontent.com/128783011/227446736-4128a01f-5448-4b5e-b2ce-dfef86098182.png)
![Snipaste_2023-03-24_14-50-46](https://user-images.githubusercontent.com/128783011/227446756-ae985691-298d-41c1-9e51-f7da5ef835b6.png)

**Comment by MrD-RC** [comment]
DSHOT is not supported on that FC in INAV. It is documented. https://github.com/iNavFlight/inav/wiki/Boards,-Targets-and-PWM-allocations#board-speedybeef405v3 and https://github.com/iNavFlight/inav/blob/master/docs/boards/SPEEDYBEEF405V3.md

**Comment by ShadowWalker8642** [comment]
dshot has issues with m4 on speedybeef405v3 i enabled dshot and was able to reconfigure mixer to use dshot

**Reply by meet-acko** [reply]
@ShadowWalker8642  How did you do it? I am not able to set dshot protocols on this board.

**Reply by MrD-RC** [reply]
DSHOT is not supported on that FC. I posted a link to the document stating that.

**Reply by cbf123** [reply]
The question is *why* isn't it supported when Betaflight and ArduCopter both support DShot on it.

**Reply by Jetrell** [reply]
@cbf123 Because INAV only uses Direct Memory Access timers to drive Dshot.. Which is the most reliable way to do it.
But due to the hardware limitation of the F405.. Concessions were made by Speedybee, in where they placed these DMA streams. e.g. SPI or Dshot.. But it appears they didn't take DMA port allocation for Dshot into account, when adding other peripherals.

There are many other manufactures like Diatone or Airbot, that have F405 controllers that support Dshot in INAV.

Betaflight and Ardupilot use a Dshot bit-bang method to work around this hardware limitation, but it does produce errors on the occasion.

The problem here, is with the manufacture not adhering to the standard INAV uses. And instead thinking all open source software platforms use the same method.

**Reply by meet-acko** [reply]
@Jetrell @MrD-RC I understand that SpeedyBee F405 v3 doesn’t support DSHOT. But their SpeedyBee BLS 50A ESC, only supports dshot. I have tried with multishot and standard protocol for this stack, but when I try to fly it afterwards, only 1 motor spins and If I increase the throttle then 2nd one and so on. Thus, drone is not lifting in the air due to imbalance whereas all four motors were spinning as expected while I was configuring the output tab. This same stack is working in betaFlight for DSHOT protocol and same issue is of motor not spinning coming in the betaflight also when using multishot protocol. 

Basically INAV is not working for SpeedyBee F405 FC stack. If there’s any way by which I can use INAV with any protocol or any hack which will solve my problem of imbalance or enable my drone to fly then please share it here.

**Reply by Jetrell** [reply]
@meet-acko I don't see a work around or short term fix for this problem. 

This is a monumental screw-up on Speedybee's part... They obviously haven't tested this hardware for INAV support.. Yet they make money from customers on false pretenses, saying it works with INAV
 
![Capture](https://github.com/iNavFlight/inav/assets/47995726/f6da881c-28d4-4d28-a5ec-5ec13a4560a2)
![Capture1](https://github.com/iNavFlight/inav/assets/47995726/b63df2ad-8964-4335-94ae-1d4903cfeb69)

I would suggest that everyone here contacts Speedybee, and point out their false information. And insist they provide you all with an ESC that works with Multishot. 
I feel your frustration.. I experienced the same last year, with the Matek F405miniTE.. And that piece of trash is still sitting around doing nothing.

**Comment by ghost** [comment]
Could this be the reason, that the copter sometimes is not able to take off? flipping at earth and doing chicken dance?
Than, this combination will not be usable?
With ESC protocol = "standard" it works partly....but only sometimes.

**Reply by DzikuVx** [reply]
No, this is not related

**Reply by ghost** [reply]
What could be the reason?

**Comment by ghost** [comment]
Could Multishot be used? Would this be possbile? Or only standard Mode can be used (analog)

**Comment by Niko-La** [comment]
is this still an issue? I not able get motor 4 and 1 to spin when using msp (they keep sputtering and reversing direction)but works fine in gui motor spinning.  the other two motors are spininning fine
what esc protocol should we use?

**Reply by MrD-RC** [reply]
This will be an issue until INAV 7.

**Comment by MrD-RC** [comment]
Fixed in INAV 7.0.
